mds/migrations/0040_map_old_to_new_event_records.py

```diff
@@ -0,0 +1,109 @@
+# Generated by Django 2.1.11 on 2019-08-16 14:27
+
+import datetime
+
+from django.db import migrations, transaction
+from django.db.models import Q
+
+from mds.provider_mapping import (
+    OLD_PROVIDER_REASON_TO_AGENCY_EVENT,
+    OLD_TO_NEW_AGENCY_EVENT,
+    PROVIDER_REASON_TO_AGENCY_EVENT,
+    get_new_event_from_old,
+    get_old_event_from_new,
+)
+
+
+def full_batch_apply(
+    function, qs, collation=lambda a, b: a and b, initial_value=None, **kwargs
+):
+    batch_size = 1000
+    try:
+        first_pk = qs.order_by("pk").values_list("pk", flat=True)[0]
+    except IndexError:
+        return initial_value
+    last_pk = qs.order_by("-pk").values_list("pk", flat=True)[0]
+    start = max(1, first_pk)
+
+    for i in range(start, last_pk + 1, batch_size):
+        queryset = qs.filter(pk__gte=i, pk__lt=min(last_pk + 1, i + batch_size))
+        try:
+            with transaction.atomic():
+                initial_value = collation(initial_value, function(queryset, **kwargs))
+        except Exception:
+            raise
+
+    return initial_value
+
+
+def fill_event_type_and_reason(apps, schema_editor):
+    EventRecord = apps.get_model("mds", "EventRecord")
+
+    for agency_event in OLD_PROVIDER_REASON_TO_AGENCY_EVENT.values():
+        if agency_event in [x[0] for x in PROVIDER_REASON_TO_AGENCY_EVENT.values()]:
+            # We do not want to apply the migration for the events
+            # that are in both mappings (service_end, service_start, trip_end and trip_start)
+            continue
```

Comment:
But also no need to compute the new `event_type_reason` field?

Reply:
By default the `event_type_reason` will be null